frontend/src/metabase/query_builder/actions/query-updates.ts

```diff
@@ -0,0 +1,21 @@
+import type { Dispatch, GetState } from "metabase-types/store";
+
+import * as MetabaseLib from "metabase-lib/v2";
+import type { Limit } from "metabase-lib/v2/types";
+
+import { getQuestion } from "../selectors";
+import { updateQuestion } from "./core";
+
+export const setLimit =
+  (limit: Limit) => (dispatch: Dispatch, getState: GetState) => {
+    const state = getState();
+    const question = getQuestion(state);
+    if (!question) {
+      return;
+    }
+    const query = question._getMLv2Query();
+    const nextQuery = MetabaseLib.limit(query, limit);
+    const nextLegacyQuery = MetabaseLib.toLegacyQuery(nextQuery);
+    const nextQuestion = question.setDatasetQuery(nextLegacyQuery);
+    dispatch(updateQuestion(nextQuestion, { run: true }));
```

Comment:
This will feel a lot better when we no longer need the `Question` wrapper and this becomes just
```ts
const query = getQuery(state);
if (query) {
  dispatch(updateQuery(MetabaseLib.limit(query, limit), { run: true }));
}
```

Reply:
Yeah, let's see how it looks once we add a few more actions here, maybe we'd generalize them for some time